js/Makefile

```diff
@@ -40,9 +40,21 @@ languages:
 engine:
 	cd engine; $(MAKE)
 
+.PHONY: pro-setup
+pro-setup:
+ifdef SEMGREP_PRO
```

Comment:
Why do we do that? What's the effect? What happens if we remove this for semgrep-pro? We get some linking error?

Reply:
Yes, a bunch of build files for the parsers and engine depend on a specific build directory, and this is easier than making changes to all of those. I'll leave a comment about this

Reply:
what would the changes look like?

Reply:
anywhere `_build` is referenced would have to by grabbed from an environment variable, as currently they are all hardcoded to find _build in `semgrep/_build`. We'd have to change some of the references in the engine files, and in the shared lang files. Imo symlinking feels less hacky than passing this as an environment variable, but I don't feel strongly. I also checked if dune will tell you the build dir location, and it will not